sql-plugin/src/main/301until320-nondb/scala/com/nvidia/spark/rapids/shims/v2/Spark30XShims.scala

```diff
@@ -50,6 +50,15 @@ trait Spark30XShims extends SparkShims {
     conf.getConf(SQLConf.LEGACY_PARQUET_REBASE_MODE_IN_READ)
   override def parquetRebaseWrite(conf: SQLConf): String =
     conf.getConf(SQLConf.LEGACY_PARQUET_REBASE_MODE_IN_WRITE)
+  override def int96ParquetRebaseRead(conf: SQLConf): String =
```

Comment:
this is in 301until320 but we have 31x shim below, should this be in 301until310?

Reply:
Yes, I think that makes more sense. The reason why this still works is because Spark31XShims will override this behavior. Depending on how expedient of a resolution is required as this is failing in nightly builds, should we handle this refactor as a follow-on?

Reply:
its fine to follow on

Reply:
sorry changed my mind because there are other issues here I think, lets get this cleaned up

Reply:
Looking at this more, as per current structure, I think this is correct because 301until310-nondb/SparkBaseShims extends 301until320-nondb/Spark30XShims. 

Won't it be better to do this is the other way where 301until320-nondb/SparkBaseShims extends 301until310-nondb/Spark30XShims? I don't know if we want to do this change as part of this PR though

Reply:
ok, I get why 30XShims might be here since everything bases on it.  We might want to come up with better name and really there is SparkBaseShims that we can likely combine, so this file is fine, but the 31XShims is under 301until320-nondb and would expect it to be only 310+

Reply:
31XShims is already in 311until320-nondb folder as part of this PR. Unless I am misreading your comment and you mean something else